packages/itwin/tree-widget/src/components/trees/stateless/imodel-content-tree/IModelContentTreeDefinition.ts

```diff
@@ -0,0 +1,701 @@
+              JOIN bis.Subject [subject] ON [subject].ECInstanceId = [partition].Parent.Id OR json_extract([subject].JsonProperties,'$.Subject.Model.TargetPartition') = printf('0x%x', [partition].ECInstanceId)
+              WHERE
+                NOT model.IsPrivate
+                AND (
+                  [subject].ECInstanceId IN (${subjectIds.map(() => "?").join(",")})
+                  OR [subject].ECInstanceId IN (
+                    SELECT s.ECInstanceId
+                    FROM child_subjects s
+                    WHERE s.RootId IN (${subjectIds.map(() => "?").join(",")}) AND s.${NodeSelectClauseColumnNames.HideNodeInHierarchy}
+                  )
+                )
+            ) childModel
+            JOIN ${modelFilterClauses.from} this ON this.ECInstanceId = childModel.ECInstanceId
+            ${modelFilterClauses.joins}
+            ${modelFilterClauses.where ? `AND (childModel.${NodeSelectClauseColumnNames.HideNodeInHierarchy} OR ${modelFilterClauses.where})` : ""}
+          `,
+          bindings: [
+            ...subjectIds.map((id): ECSqlBinding => ({ type: "id", value: id })),
+            ...subjectIds.map((id): ECSqlBinding => ({ type: "id", value: id })),
+          ],
+        },
+      },
+    ];
+  }
+
+  private async createISubModeledElementChildrenQuery({
+    parentNodeInstanceIds: elementIds,
+  }: DefineInstanceNodeChildHierarchyLevelProps): Promise<HierarchyLevelDefinition> {
+    // note: we do not apply hierarchy level filtering on this hierarchy level, because it's always
+    // hidden - the filter will get applied on the child hierarchy levels
+    return [
+      {
+        fullClassName: "BisCore.Model",
+        query: {
+          ecsql: `
+            SELECT
+              ${await this._selectQueryFactory.createSelectClause({
+                ecClassId: { selector: "this.ECClassId" },
+                ecInstanceId: { selector: "this.ECInstanceId" },
+                nodeLabel: "", // doesn't matter - the node is always hidden
+                hideNodeInHierarchy: true,
+              })}
+            FROM BisCore.Model this
+            WHERE
+              this.ModeledElement.Id IN (${elementIds.map(() => "?").join(",")})
+              AND NOT this.IsPrivate
+          `,
+          bindings: [...elementIds.map((id): ECSqlBinding => ({ type: "id", value: id }))],
+        },
+      },
+    ];
+  }
+
+  private async createGeometricModelChildrenQuery({
+    parentNodeInstanceIds: modelIds,
+    instanceFilter,
+    viewType,
+  }: DefineInstanceNodeChildHierarchyLevelProps & { viewType: "2d" | "3d" }): Promise<HierarchyLevelDefinition> {
+    const { categoryClass, elementClass } = getClassNameByViewType(viewType);
+    const instanceFilterClauses = await this._selectQueryFactory.createFilterClauses({
+      filter: instanceFilter,
+      contentClass: { fullName: categoryClass, alias: "this" },
```

Comment:
I think this means we won't be able to filter `bis.InformationContentElement` nodes?

Reply:
I remember fixing it, but it seems like I accidentally removed the change, fixed.